coordinator/tributary/src/blockchain.rs

```diff
@@ -166,6 +166,16 @@ impl<D: Db, T: TransactionTrait> Blockchain<D, T> {
     self.provided.provide(tx)
   }
 
+  pub(crate) fn complete_provided(&mut self, order: &str, hash: [u8; 32]) {
+    let mut txn = self.db.as_mut().unwrap().txn();
+    self.provided.complete(&mut txn, order, hash);
+    txn.commit();
+  }
+
+  pub(crate) fn provided_transactions(&self) -> Vec<u8> {
```

Comment:
While it's irrelevant, due to my above commentary on how this *should* be handled, we shouldn't return a raw byte buffer with no meaning no context. The Tributary APIs should be self contained and safe, meaning this should've at least been a `Vec<[u8; 32]>`.